worker_test.go

```diff
@@ -84,6 +123,52 @@ func testWorkerStart(t *testing.T, connPool adapter.ConnPool) {
 	assert.False(t, w.running)
 }
 
+func TestWorkerPool_Run(t *testing.T) {
+	t.Run("pgx/v3", func(t *testing.T) {
+		testWorkerPoolRun(t, adapterTesting.OpenTestPoolPGXv3(t))
+	})
+	t.Run("pgx/v4", func(t *testing.T) {
+		testWorkerPoolRun(t, adapterTesting.OpenTestPoolPGXv4(t))
+	})
+	t.Run("lib/pq", func(t *testing.T) {
+		testWorkerPoolRun(t, adapterTesting.OpenTestPoolLibPQ(t))
+	})
+}
+
+func testWorkerPoolRun(t *testing.T, connPool adapter.ConnPool) {
+	c := NewClient(connPool)
+
+	poolSize := 2
+	w := NewWorkerPool(c, WorkMap{}, poolSize)
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	var grp errgroup.Group
+	grp.Go(func() error {
+		return w.Run(ctx)
+	})
+
+	// give worker time to start
+	time.Sleep(time.Second)
+
+	assert.True(t, w.running)
+	for i := range w.workers {
+		assert.True(t, w.workers[i].running)
+	}
+
+	// try to start one more time to get an error about already running worker pool
+	assert.Error(t, w.Run(context.Background()))
+
+	cancel()
+
+	assert.NoError(t, grp.Wait())
```

Comment:
If this condition fails - it does not make sense to continue the test flow

```suggestion
	require.NoError(t, grp.Wait())
```